qcfractal/alembic/versions/4bb79efa9855_add_queue_manager_id_to_base_results.py

```diff
@@ -0,0 +1,30 @@
+"""Add queue_manager id to Base_results
+
+Revision ID: 4bb79efa9855
+Revises: 26cfd7b0439e
+Create Date: 2019-07-26 15:31:14.811337
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '4bb79efa9855'
+down_revision = '26cfd7b0439e'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('base_result', sa.Column('manager_name', sa.String(), nullable=True))
+    op.create_foreign_key(None, 'base_result', 'queue_manager', ['manager_name'], ['name'], ondelete='SET NULL')
```

Comment:
Just checking did we wanted to skip making this a foreign key for speed as we are less concerned with consistency here.

Reply:
It's a foreign key using a "unique" non primary key. This is valid.

Reply:
Right, just curious if we wanted to not add another FK as the results have many FKs on them already which will potentially slow down the inserts.

Reply:
Well, it's a compromise. If you think it's going to be okay missing up with the manager_name, and will be always forgiving about it then it's okay to remove the FK constraint.
If we remove the FK, and if yo delete a manager, then the manager_name will still be there, referncing wrong ID. So, anytime you look for those managers, you put this in mind.
Also it can happen that you remove a manger and add another with the same name or ID.

In short, it's the integrity issue.

I think since we only touch the PK constraint once (when marking results as complete), it won't be a huge time issue. This constraint won't affect reads.
So I suggest to keep it.

Reply:
Ok, sounds good to me. Just wanted to double check since we chatted about it.